From 3f146445166d2c65d39fb312c983a55247a8fd1d Mon Sep 17 00:00:00 2001
From: Will Dietz <w@wdtz.org>
Date: Fri, 25 Mar 2022 15:13:10 -0500
Subject: [PATCH] llvm/cmake: fix installation destination for standalone MLIR

This isn't suitable for normal LLVM use, as that should be installed
into LLVM_TOOLS_INSTALL_DIR (although they're often the same).

However, installing to LLVM's tool directory from a standalone MLIR
build is problematic -- MLIR's bits go in its own bin directory.
---
 llvm/cmake/modules/TableGen.cmake | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/llvm/cmake/modules/TableGen.cmake b/llvm/cmake/modules/TableGen.cmake
index 59fd0d3b0733..d0f0bd8da170 100644
--- a/llvm/cmake/modules/TableGen.cmake
+++ b/llvm/cmake/modules/TableGen.cmake
@@ -192,7 +192,7 @@ macro(add_tablegen target project)
     install(TARGETS ${target}
             ${export_to_llvmexports}
             COMPONENT ${target}
-            RUNTIME DESTINATION ${LLVM_TOOLS_INSTALL_DIR})
+            RUNTIME DESTINATION "${CMAKE_INSTALL_BINDIR}")
     if(NOT LLVM_ENABLE_IDE)
       add_llvm_install_targets("install-${target}"
                                DEPENDS ${target}
-- 
2.35.1

